packages/contracts-bedrock/scripts/libraries/DeployUtils.sol

```diff
@@ -195,6 +202,68 @@ library DeployUtils {
         return create2AndSave(_save, _name, _name, hex"", _salt);
     }
 
+    /// @notice Deploys a contract with the given name using CREATE2. If the contract is already deployed, this method
+    /// does nothing.
+    /// @param _name Name of the contract to deploy.
+    /// @param _args ABI-encoded constructor arguments.
+    function createDeterministic(
+        string memory _name,
+        bytes memory _args,
+        bytes32 _salt
+    )
+        internal
+        returns (address payable addr_)
+    {
+        bytes memory initCode = abi.encodePacked(vm.getCode(_name), _args);
+        address preComputedAddress = vm.computeCreate2Address(_salt, keccak256(initCode));
+        if (preComputedAddress.code.length > 0) {
+            addr_ = payable(preComputedAddress);
+        } else {
+            vm.broadcast(msg.sender);
```

Comment:
Is `vm.broadcast` required here? I didn't think it would be

Reply:
I think it makes more sense to keep broadcasting logic outside of this function

Reply:
ok refactored this out